runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -409,6 +428,46 @@ function LanguageTree:_add_injections()
   return query_time
 end
 
+--- Run an asynchronous parse, calling {on_parse} when complete.
+---
+--- @param range boolean|Range|nil
+--- @param on_parse fun(trees: TSTrees)
+function LanguageTree:_async_parse(range, on_parse)
+  coroutine.resume(
+    --- @param co_range boolean|Range|nil
+    --- @param co_cb fun(trees: TSTrees)
+    coroutine.create(function(co_range, co_cb)
+      local co = coroutine.running()
+      local ct = self:_buf().changedtick
+      local unfinished = true
```

Comment:
Would it be simply to remove this and just break from the loop when it's finished?